File: LinkedList_2.c
#include<stdio.h>
#include<stdlib.h>

typedef struct node
{
	int value;
	struct node* next;
}node;

node* head;

node* createnode(int val,node* nxt)
{
	node* m=malloc(sizeof(node));
	m->value=val;
	m->next=nxt;
	return m;
}

void insert(int val)
{
	head=createnode(val,head);
}

void insertat(int val,int idx)
{
	node* p=head;
	int i;
	if(idx==1)
	{
		insert(val);
		return;
	}
	for(i=0;i<idx-2;i++)
	{
		if(p!=NULL)
			p=p->next;
		else
		{
			printf("Index is invalid\n");
			return;
		}
	}
	p->next=createnode(val,p->next);
}

void insertMultiple(int n,int a[])
{
	int i;
	for(i=0;i<n;i++)
	{
		insert(a[i]);
	}
}

void delete()
{
	head=head->next;
}

void deleteAt(int idx)
{
	if(idx==1)
	{
		delete();
		return;
	}
	node* p=head;
	int i;
	for(i=0;i<idx-2;i++)
	{
		if(p!=NULL)
			p=p->next;
		else
		{
			printf("Index is Invalid\n");
			return;
		}
	}
	p->next=p->next->next;
	return;
}

void display()
{
	node* p=head;
	while(p!=NULL)
	{
		printf("%d -> ",p->value);
		p=p->next;
	}
	printf("\n");
}

void main()
{
	int i=1;
	while(i!=0)
	{
		printf("1. insert \n2. insertAt\n3. insert Multiple\n4. delete\n5. deleteAt\n6. display\n0. exit\ni=");
		scanf("%d",&i);
		int v;
		if(i==1)
		{
			printf("Enter the value\n");
			scanf("%d",&v);
			insert(v);
		}
		else if(i==2)
		{
			int val,idx;
			printf("insert val and index\n");
			scanf("%d %d",&val,&idx);
			insertat(val,idx);
		}
		else if(i==3)
		{
			printf("Enter the size of inputs\n");
			int n;
			scanf("%d",&n);
			int *a=malloc(4*n);
			printf("Enter the inputs\n");
			int i;
			for(i=0;i<n;i++)
			{
				scanf("%d",&a[i]);
			}
			insertMultiple(n,a);
		}
		else if(i==4)
		{
			delete();
		}
		else if(i==5)
		{
			int idx;
			printf("insert index\n");
			scanf("%d",&idx);
			deleteAt(idx);
		}
		else if(i==6)
			display();
		else
			return;
		printf("\n");
	}
}

File: PROGRAMS/Python/LinkedList_2.py
class node:
    def __init__(self,val=None,nxt=None):
        self.value=val
        self.next=nxt


class LinkedList:
    def __init__(self):
        self.head=node()
    def insert(self,x,pre):
        m=node(x,pre.next)
        pre.next=m
    def display(self):
        p=self.head
        while(p.next!=None):
            print(p.next.value,end=" -> ")
            p=p.next
        print()
    def insertAt(self,x,i):
        p=self.head
        for i in range(i-1):
            if p.next!=None:
                p=p.next
            else:
                printf("Wrong index");
        self.insert(x,p)

def main():
    i=1
    L=LinkedList()
    while i!=3:
        print("Choose 1.insertAt 2.display 3.close")
        i=int(input())
        if i==1:
            k=int(input("Enter index  "))
            l=input("Enter data  ")
            L.insertAt(l,k)
        elif i==2:
            L.display()
        elif i==3:
            break
if __name__=='__main__':
    main()



            
